index.d.ts

```diff
@@ -0,0 +1,182 @@
+// Type definitions for redux-little-router 14.0.0
+// Project: https://github.com/FormidableLabs/redux-little-router
+// Definitions by: priecint <https://github.com/priecint>
+
+import * as React from "react";
+import {Reducer, Middleware, StoreEnhancer} from "redux";
+
+export interface Routes {
+  [index: string]: {
+    [index: string]: RouteDefinition;
+  };
+}
+
+export interface RouteDefinition {
+  [index: string]: string | RouteDefinition;
+}
+
+export type LocationOptions = {
+  persistQuery?: boolean;
+};
+
+export type Query = { [key: string]: string };
+export type Params = { [key: string]: string };
+
+export type Location = {
+  basename?: string;
+  options?: LocationOptions;
+  params?: Params;
+  previous?: Location;
+  query?: Query;
+  result?: {};
+};
+
+interface BrowserRouterOptions {
+  routes: Routes;
+  basename?: string;
+}
+
+export function routerForBrowser(options: BrowserRouterOptions): {
+  reducer: Reducer<Location>;
+  middleware: Middleware;
+  enhancer: StoreEnhancer<Location>;
+};
+
+interface ServerRouterOptions {
+  routes: Routes;
+  request: {
+    path: string;
+    baseUrl: string;
+    url: string;
+    query: {
+      [key: string]: string;
+    }
+  };
+}
+
+export function routerForExpress(options: ServerRouterOptions): {
+  reducer: Reducer<Location>;
+  middleware: Middleware;
+  enhancer: StoreEnhancer<Location>;
+};
+
+interface HapiRouterOptions {
+  routes: Routes;
+  request: {
+    path: string;
+    url: string;
+    query: {
+      [key: string]: string;
+    }
+  };
+}
+
+export function routerForHapi(options: HapiRouterOptions): {
+  reducer: Reducer<Location>;
+  middleware: Middleware;
+  enhancer: StoreEnhancer<Location>;
+};
+
+export function initializeCurrentLocation(location: Location): {
+  type: string;
```

Comment:
Actions should be using string constant types (like `type: 'ROUTER_PUSH'`) instead of just `string` as type declaration. Otherwise this can't be used in places where actions are strongly typed.